src/Microsoft.FeatureManagement.AspNetCore/TagHelpers/FeatureTagHelper.cs

```diff
@@ -52,16 +59,35 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
 
             if (!string.IsNullOrEmpty(Name))
             {
-                IEnumerable<string> names = Name.Split(',').Select(n => n.Trim());
+                IEnumerable<string> features = Name.Split(',').Select(n => n.Trim());
 
-                enabled = Requirement == RequirementType.All ?
-                    await names.All(async n => await _featureManager.IsEnabledAsync(n).ConfigureAwait(false)) :
-                    await names.Any(async n => await _featureManager.IsEnabledAsync(n).ConfigureAwait(false));
-            }
+                if (string.IsNullOrEmpty(Variant))
+                {
+                    enabled = Requirement == RequirementType.All ?
+                        await features.All(async feature => await _featureManager.IsEnabledAsync(feature).ConfigureAwait(false)) :
+                        await features.Any(async feature => await _featureManager.IsEnabledAsync(feature).ConfigureAwait(false));
 
-            if (Negate)
-            {
-                enabled = !enabled;
+                    if (Negate)
+                    {
+                        enabled = !enabled;
+                    }
+                }
+                else
+                {
+                    if (features.Count() != 1)
+                    {
+                        throw new ArgumentException("Variant cannot be associated with multiple feature flags.", nameof(Name));
+                    }
+
+                    IEnumerable<string> variants = Variant.Split(',').Select(n => n.Trim());
+
+                    enabled = await variants.Any(
+                        async variant => {
+                            Variant assignedVariant = await _featureManager.GetVariantAsync(features.First());
+
+                            return variant == assignedVariant?.Name;    
+                        });
+                }
```

Comment:
I think we should support Negate in the variants case too.

Reply:
I can think of scenarios where variant negate can be useful.

Reply:
Negate can be treated as just reverse the original behavior of the tag helper. In this case, I agree Negate can also apply to variant.

Reply:
Normal Feature Flag Negate:
```
<feature negate="true" name="ShouldEnable">
  <p>This can only be seen if 'ShouldEnable' is not enabled.</p>
</feature>
```

Variant Feature Flag Negate
```
<feature negate="true" name=”Alpha” variant=”V1”>
  <p>This can only be seen if 'Alpha' is not V1.</p>
</feature>
```

Reply:
Ah, it does make sense.